### ISSUE TITLE
Marketplace Checkout - Crypto transactions

### ISSUE THREAD

**Issue by BellringerQuinn** [0xsequence/sequence-unreal#244]
Depends on #228 

Implement marketplace checkout methods, namely:

- GenerateBuyTransaction
- GenerateSellTransaction
- GenerateListingTransaction
- GenerateOfferTransaction

Ridl file: https://github.com/0xsequence/marketplace-api/blob/master/proto/marketplace.ridl#L605
Unity Implementation:

- https://github.com/0xsequence/sequence-unity/blob/Feature/marketplaceCheckout/Packages/Sequence-Unity/Sequence/SequenceSDK/Marketplace/Checkout.cs and its dependencies 
- https://github.com/0xsequence/sequence-unity/blob/Feature/marketplaceCheckout/Assets/SequenceSDK/Marketplace/MarketplaceCheckoutTests.cs these tests can be helpful to give you a general idea of how the flow works, specifically TestCreateListingAndBuyIt and TestCreateOfferAndSellIt

In general, the way the marketplace works is as follows:

- If you have an ERC721/1155 you wish to sell, you can either a) create a listing or b) accept an existing offer
- If you wish to buy an ERC721/1155, you can either a) create an offer or b) accept an existing listing

Creating a listing or offer is a non-instant way of selling or buying an ERC721/1155; they require someone on the other end to accept them.
Accepting a listing or offer via a buy or sell operation respectively, is an instant way of selling or buying an ERC721/1155. To do this, you must query for active listings or offers using the marketplace read methods.

In other words, every buy/sell is a two sided operation. One person will list an ERC721/1155 for sale or make an offer to buy one; the other party will buy the listing or sell to the first person's offer.

All four API methods will return an array of "Steps". These Steps are essentially transactions. Using the GenerateListingTransaction endpoint alone will not create the listing, you must take the Step[] response and convert those into a Transaction[] to then be submitted via the WaaS API for the Listing to be created. The same logic applies for the other API methods above.
